wandb/sdk/service/server_sock.py

```diff
@@ -141,6 +143,50 @@ def server_inform_start(self, sreq: "spb.ServerRequest") -> None:
         settings = settings_dict_from_pbmap(request._settings_map)
         self._mux.update_stream(stream_id, settings=settings)
 
+    def server_inform_console_data(self, sreq: "spb.ServerRequest") -> None:
+        request = sreq.inform_console_data
+        for run_id in self._console_run_ids:
+
+            name = request.output_type
+            data = request.output_data
+            if name == "stdout":
+                otype = pb.OutputRecord.OutputType.STDOUT
+            elif name == "stderr":
+                otype = pb.OutputRecord.OutputType.STDERR
+            else:
+                assert False
+
+            o = pb.OutputRecord(output_type=otype, line=data)
+            o.timestamp.GetCurrentTime()
+            
+            record = pb.Record()
+            record.output.CopyFrom(o)
+
+            stream_id = run_id
+            # print(f"SEND: {stream_id}, {record}")
+            try:
+                iface = self._mux.get_stream(stream_id).interface
+            except Exception as e:
+                # TODO this might happen at shutdown, handle this better
+                continue
+            assert iface.record_q
+            iface.record_q.put(record)
+
+
+    def server_inform_console_start(self, sreq: "spb.ServerRequest") -> None:
+        request = sreq.inform_console_start
+        run_id = request.run_id
+        print(f"DEBUG console start {run_id}")
+        # TODO: move this datastructure to mux?
```

Comment:
yeah... i think it would be better... so we have this logic handled the same way for grpc